cpp/cmake_modules/DefineOptions.cmake

```diff
@@ -109,6 +109,9 @@ if(ARROW_DEFINE_OPTIONS)
 
   define_option(ARROW_NO_DEPRECATED_API "Exclude deprecated APIs from build" OFF)
 
+  define_option(ARROW_POSITION_INDEPENDENT_CODE
```

Comment:
Shall we disable `ARROW_BUILD_SHARED` if PIC is off (or fail cmake configuration)? I see many link errors otherwise.

Reply:
Good idea!
But I disable `ARROW_POSITION_INDEPENDENT_CODE` (`ARROW_BUILD_SHARED=ON` and `ARROW_POSITION_INDEPENDENT_CODE=ON`) with `ARROW_BUILD_SHARED=ON` and `ARROW_POSITION_INDEPENDENT_CODE=OFF` instead of disabling `ARROW_BUILD_SHARED` (`ARROW_BUILD_SHARED=OFF` and `ARROW_POSITION_INDEPENDENT_CODE=OFF`). Because `ARROW_BUILD_SHARED=OFF` causes other errors with other options such as `ARROW_FLIGHT=ON`.